### DISCUSSION TITLE
framework-espidf @ 3.50301.0 (5.3.1)

### DISCUSSION THREAD

**Discussion by tv4you2016** [schreibfaul1/ESP32-audioI2S#861] [Q&A] [answered]
[env:esp32dev]
platform = espressif32
board = esp32dev
framework = espidf
monitor_speed = 115200  ; 921600, 512000, 460800, 256000, 115200
board_build.f_cpu = 240000000L
board_build.flash_freq=80M
board_upload.flash_size=4MB
board_build.spiram_mode=2
board_build.flash_mode = qio
board_build.bootloader = dio
monitor_filters  = 
    direct
    esp32_exception_decoder
board_build.partitions = partitions.csv
upload_speed = 921600

> Verbose mode can be enabled via `-v, --verbose` option
> CONFIGURATION: https://docs.platformio.org/page/boards/espressif32/esp32dev.html
> PLATFORM: Espressif 32 (6.9.0) > Espressif ESP32 Dev Module
> HARDWARE: ESP32 240MHz, 320KB RAM, 4MB Flash
> DEBUG: Current (cmsis-dap) External (cmsis-dap, esp-bridge, esp-prog, iot-bus-jtag, jlink, minimodule, olimex-arm-usb-ocd, olimex-arm-usb-ocd-h, olimex-arm-usb-tiny-h, olimex-jtag-tiny, tumpa)
> PACKAGES: 
>  - framework-espidf @ 3.50301.0 (5.3.1) 
>  - tool-cmake @ 3.16.4 
>  - tool-esptoolpy @ 1.40501.0 (4.5.1) 
>  - tool-idf @ 1.0.1 
>  - tool-mconf @ 1.4060000.20190628 (406.0.0) 
>  - tool-ninja @ 1.9.0 
>  - tool-riscv32-esp-elf-gdb @ 11.2.0+20220823 
>  - tool-xtensa-esp-elf-gdb @ 11.2.0+20230208 
>  - toolchain-esp32ulp @ 1.23800.240113 (2.38.0) 
>  - toolchain-xtensa-esp-elf @ 13.2.0+20240530
> Reading CMake configuration...
> -- Found Git: C:/Program Files/Git/cmd/git.exe (found version "2.46.2.windows.1") 
> -- git rev-parse returned 'fatal: not a git repository (or any of the parent directories): .git'
> -- The C compiler identification is GNU 13.2.0
> -- The CXX compiler identification is GNU 13.2.0
> -- The ASM compiler identification is GNU
> -- Found assembler: C:/Users/Asus/.platformio/packages/toolchain-xtensa-esp-elf/bin/xtensa-esp32-elf-gcc.exe
> -- Check for working C compiler: C:/Users/Asus/.platformio/packages/toolchain-xtensa-esp-elf/bin/xtensa-esp32-elf-gcc.exe
> -- Check for working C compiler: C:/Users/Asus/.platformio/packages/toolchain-xtensa-esp-elf/bin/xtensa-esp32-elf-gcc.exe -- works
> -- Detecting C compiler ABI info
> -- Detecting C compiler ABI info - done
> -- Detecting C compile features
> -- Detecting C compile features - done
> -- Check for working CXX compiler: C:/Users/Asus/.platformio/packages/toolchain-xtensa-esp-elf/bin/xtensa-esp32-elf-g++.exe
> -- Check for working CXX compiler: C:/Users/Asus/.platformio/packages/toolchain-xtensa-esp-elf/bin/xtensa-esp32-elf-g++.exe -- works
> -- Detecting CXX compiler ABI info
> -- Detecting CXX compiler ABI info - done
> -- Detecting CXX compile features
> -- Detecting CXX compile features - done
> -- Could not use 'git describe' to determine PROJECT_VER.
> -- Building ESP-IDF components for target esp32
> NOTICE: Skipping optional dependency: espressif/esp-sr
> 
> Manifest files have changed, solving dependencies.
> 
> ...........................................................................HINT: Please check manifest file of the following component(s): arduino
> 
> -- Configuring incomplete, errors occurred!
> See also "C:/Users/Asus/Documents/PlatformIO/Projects/_iSND1/.pio/build/esp32dev/CMakeFiles/CMakeOutput.log".
> 
> fatal: not a git repository (or any of the parent directories): .git
> CMake Error at .pio/build/esp32dev/CMakeFiles/git-data/grabRef.cmake:48 (file):
>   file failed to open for reading (No such file or directory):
> 
>     C:/Users/Asus/Documents/PlatformIO/Projects/_iSND1/.pio/build/esp32dev/CMakeFiles/git-data/head-ref
> Call Stack (most recent call first):
>   C:/Users/Asus/.platformio/packages/framework-espidf/tools/cmake/third_party/GetGitRevisionDescription.cmake:83 (include)
>   C:/Users/Asus/.platformio/packages/framework-espidf/tools/cmake/third_party/GetGitRevisionDescription.cmake:93 (get_git_head_revision)
>   C:/Users/Asus/.platformio/packages/framework-espidf/tools/cmake/project.cmake:693 (git_describe)
>   CMakeLists.txt:3 (project)
> 
> 
> CMake Error at C:/Users/Asus/.platformio/packages/framework-espidf/tools/cmake/build.cmake:552 (message):
>   WARNING: Component "idf" not found
> 
>   ERROR: Because project depends on idf (>=5.1,<5.2) which doesn't match any
>   versions, version solving failed.
> 
> Call Stack (most recent call first):
>   C:/Users/Asus/.platformio/packages/framework-espidf/tools/cmake/project.cmake:710 (idf_build_process)
>   CMakeLists.txt:3 (project)
> 
> 
> 
> ========================= [FAILED] Took 30.06 seconds =========================

how to solved?

**Comment by schreibfaul1** [comment, marked as answer]
framework = espidf is not enough, the Arduino libraries, such as WiFiClient, NetworkClient... are used
maybe like this:
````
[env]
platform = https://github.com/pioarduino/platform-espressif32/releases/download/53.03.10-rc1/platform-espressif32.zip ;Arduino 3.1.0 / IDF 5.3.0

framework = arduino, espidf

monitor_speed   = 115200
monitor_filters = esp32_exception_decoder, send_on_enter, log2file
upload_speed    = 921600  ; 921600, 512000, 460800, 256000, 115200

board_build.embed_txtfiles =
    managed_components/espressif__esp_insights/server_certs/https_server.crt
    managed_components/espressif__esp_rainmaker/server_certs/rmaker_mqtt_server.crt
    managed_components/espressif__esp_rainmaker/server_certs/rmaker_claim_service_server.crt
    managed_components/espressif__esp_rainmaker/server_certs/rmaker_ota_server.crt
````
